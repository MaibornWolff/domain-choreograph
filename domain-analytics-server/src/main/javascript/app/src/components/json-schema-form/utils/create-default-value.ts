import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';

const defaultValues: Record<Exclude<JSONSchema4TypeName, 'object'>, any> = {
  string: undefined,
  number: undefined,
  integer: undefined,
  boolean: undefined,
  array: [],
  'null': null,
  any: ''
};

export function createDefaultValue(schema: JSONSchema4): any {
  const type = schema.type || 'any';
  if (type instanceof Array) {
    throw new Error('Array types are not supported');
  } else if (type === 'object') {
    return createDefaultValueForObject(schema);
  } else {
    return defaultValues[type];
  }
}

function createDefaultValueForObject(schema: JSONSchema4): Promise<object> {
  if (schema.type !== 'object') {
    throw new Error('Expect schema to be an object');
  }
  let defaultValue: any = {};
  if (schema.properties) {
    Object.entries(schema.properties).forEach(([key, property]) => {
      defaultValue[key] = createDefaultValue(property)
    });
  }
  return defaultValue;
}

