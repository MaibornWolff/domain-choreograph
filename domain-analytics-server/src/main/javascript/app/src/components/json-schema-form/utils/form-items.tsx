import { styled } from '~utils/styled';

export const FormItems = styled('div')`
    display: flex;
    flex-direction: column;

    & > * {
      margin-bottom: .5rem;
    }
`;
