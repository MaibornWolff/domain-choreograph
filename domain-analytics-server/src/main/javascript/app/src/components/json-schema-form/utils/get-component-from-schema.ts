import { FieldComponent } from 'clean-forms';
import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
import { DatePickerMsField } from '~components/json-schema-form/form-fields/date-picker-ms-field';
import { NumberTextField, TextField } from '~components/json-schema-form/form-fields/text-field';
import { JsonSchemaFormFieldProps } from '~components/json-schema-form/models/json-schema-form-field-props';

export interface JsonSchemaComponentMap {
  matcher?: { type: JSONSchema4TypeName, format?: string };
  field: FieldComponent<any, JsonSchemaFormFieldProps>;
}

const defaultMapping: JsonSchemaComponentMap[] = [
  { matcher: { type: 'integer', format: 'utc-millisec' }, field: DatePickerMsField },
  { matcher: { type: 'integer' }, field: NumberTextField },
  { matcher: { type: 'number' }, field: NumberTextField },
  { field: TextField }
];

export function getComponentFromSchema(
  schema: JSONSchema4,
  mapping: JsonSchemaComponentMap[] = defaultMapping
): FieldComponent<any, JsonSchemaFormFieldProps> {
  for (const map of mapping) {
    if (map.matcher == null) {
      return map.field;
    }
    const typeIsMatching = map.matcher.type === schema.type;
    const formatIsMatching = map.matcher.format == null || map.matcher.format === schema.format;
    if (typeIsMatching && formatIsMatching) {
      return map.field;
    }
  }
  throw new Error('Couldn\'t match the schema');
}
