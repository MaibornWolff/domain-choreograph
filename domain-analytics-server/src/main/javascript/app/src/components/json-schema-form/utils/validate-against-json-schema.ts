import { ValidationFunction } from 'clean-forms';
import { JSONSchema4 } from 'json-schema';
import { Validator } from 'jsonschema';


export function validateAgainstJsonSchema(schema: JSONSchema4): ValidationFunction {
  const validator = new Validator();
  return ({ value }) => {
    const result = validator.validate(value, schema as any);
    const errors = result.errors.map(error => {
      let message = error.message;
      let property = error.property
        .replace(/^instance\.?/, '');
      let schema = error.schema as JSONSchema4;
      if (error.name === 'required') {
        message = `is required`;
        schema = schema.properties![error.argument];
        property = joinProperties(property, error.argument);
      }

      const title = schema.title || '';
      message = `${title} ${message}`;

      return [ property, message ] as [string, string];
    });
    return errors;
  }
}

function joinProperties(path1: string, path2: string): string {
  if (path1 === '') {
    return path2;
  }
  if (path2 === '') {
    return path1;
  }
  return `${path1}.${path2}`;
}
