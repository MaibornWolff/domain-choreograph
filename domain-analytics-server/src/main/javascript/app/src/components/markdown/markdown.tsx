import 'katex/dist/katex.css';
import MarkdownIt from 'markdown-it';
import MarkdownItKatex from 'markdown-it-katex';
import React from 'react';
import { MarkdownStyleWrapper } from '~components/markdown/markdown-style-wrapper';
import { styled } from '~utils/styled';

const MarkdownWrapper = styled('div')`
  font-size: 1rem;
  line-height: 1rem;
  position: relative;
  font-family: inherit;
  all: inherit;
`;

export interface MarkdownProps {
  markdown: string;
}

export interface MarkdownState {
}

export class Markdown extends React.Component<MarkdownProps, MarkdownState> {
  private md: MarkdownIt.MarkdownIt = new MarkdownIt();

  public render() {
    const html: string = this.transformMarkdownToHtml(this.props.markdown);
    return (
      <MarkdownStyleWrapper>
        <MarkdownWrapper className="markdown-body" dangerouslySetInnerHTML={{ __html: html }}/>
      </MarkdownStyleWrapper>
    );
  }

  public componentWillMount() {
    this.initializeMarkdownIt();
  }

  private transformMarkdownToHtml(markdown: string): string {
    return this.md.render(markdown);
  }

  private initializeMarkdownIt(): void {
    this.md.use(MarkdownItKatex);
  }
}
