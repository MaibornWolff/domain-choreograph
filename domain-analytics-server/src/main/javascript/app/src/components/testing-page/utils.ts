export function escapeJavaClass(javaClass: string) {
  return javaClass.replace(/\./g, '-');
}

export function reverseJavaClassEscape(javaClass: string) {
  return javaClass.replace(/-/g, '.');
}
