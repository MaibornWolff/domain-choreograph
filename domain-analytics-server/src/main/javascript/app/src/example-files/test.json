{
  "nodes": {
    "A": {
      "id": "A",
      "name": "A",
      "scope": "root"
    }
  },
  "dependencies": [],
  "scopes": {
    "root": {
      "id": "root",
      "executionContext": "Application",
      "nodes": [
        "A"
      ]
    }
  },
  "executionContexts": {
    "Application": {
      "id": "Application",
      "scopes": [
        "root"
      ]
    }
  }
}
