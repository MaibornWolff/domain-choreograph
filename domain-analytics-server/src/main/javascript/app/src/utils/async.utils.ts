export function wait(milliseconds: number): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    setTimeout(() => resolve(), milliseconds);
  });
}
