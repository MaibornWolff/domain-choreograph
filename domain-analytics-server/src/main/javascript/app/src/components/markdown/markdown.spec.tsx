import { shallow } from 'enzyme';
import * as React from 'react';
import { renderWithProviders } from '~utils/test.utils';
import { Markdown } from './markdown';

describe('Markdown', () => {
  it('should render', () => {
    const markdown = `
# Hello World

This is a test

This is a **bold** test

This is a *italic* test

Should support inline latex $1 + 2 = 10$

$$
x = x_2
$$
    `;
    expect(renderWithProviders(<Markdown markdown={markdown}/>)).toMatchSnapshot();
  });
});
