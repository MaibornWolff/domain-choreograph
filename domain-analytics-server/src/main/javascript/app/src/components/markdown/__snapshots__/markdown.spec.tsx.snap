// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown should render 1`] = `
.emotion-2 .markdown-body {
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  line-height: 1.5;
  color: rgba(0,0,0,.87);
  font-size: 16px;
  line-height: 1.5;
  word-wrap: break-word;
}

.emotion-2 .markdown-body .pl-c {
  color: rgba(0,0,0,.54);
}

.emotion-2 .markdown-body .pl-c1,
.emotion-2 .markdown-body .pl-s .pl-v {
  color: #005cc5;
}

.emotion-2 .markdown-body .pl-e,
.emotion-2 .markdown-body .pl-en {
  color: #6f42c1;
}

.emotion-2 .markdown-body .pl-smi,
.emotion-2 .markdown-body .pl-s .pl-s1 {
  color: rgba(0,0,0,.87);
}

.emotion-2 .markdown-body .pl-ent {
  color: #22863a;
}

.emotion-2 .markdown-body .pl-k {
  color: #d73a49;
}

.emotion-2 .markdown-body .pl-s,
.emotion-2 .markdown-body .pl-pds,
.emotion-2 .markdown-body .pl-s .pl-pse .pl-s1,
.emotion-2 .markdown-body .pl-sr,
.emotion-2 .markdown-body .pl-sr .pl-cce,
.emotion-2 .markdown-body .pl-sr .pl-sre,
.emotion-2 .markdown-body .pl-sr .pl-sra {
  color: #032f62;
}

.emotion-2 .markdown-body .pl-v,
.emotion-2 .markdown-body .pl-smw {
  color: #e36209;
}

.emotion-2 .markdown-body .pl-bu {
  color: #b31d28;
}

.emotion-2 .markdown-body .pl-ii {
  color: #fafbfc;
  background-color: #b31d28;
}

.emotion-2 .markdown-body .pl-c2 {
  color: #fafbfc;
  background-color: #d73a49;
}

.emotion-2 .markdown-body .pl-c2::before {
  content: "^M";
}

.emotion-2 .markdown-body .pl-sr .pl-cce {
  font-weight: bold;
  color: #22863a;
}

.emotion-2 .markdown-body .pl-ml {
  color: #735c0f;
}

.emotion-2 .markdown-body .pl-mh,
.emotion-2 .markdown-body .pl-mh .pl-en,
.emotion-2 .markdown-body .pl-ms {
  font-weight: bold;
  color: #005cc5;
}

.emotion-2 .markdown-body .pl-mi {
  font-style: italic;
  color: rgba(0,0,0,.87);
}

.emotion-2 .markdown-body .pl-mb {
  font-weight: bold;
  color: rgba(0,0,0,.87);
}

.emotion-2 .markdown-body .pl-md {
  color: #b31d28;
  background-color: #ffeef0;
}

.emotion-2 .markdown-body .pl-mi1 {
  color: #22863a;
  background-color: #f0fff4;
}

.emotion-2 .markdown-body .pl-mc {
  color: #e36209;
  background-color: #ffebda;
}

.emotion-2 .markdown-body .pl-mi2 {
  color: #f6f8fa;
  background-color: #005cc5;
}

.emotion-2 .markdown-body .pl-mdr {
  font-weight: bold;
  color: #6f42c1;
}

.emotion-2 .markdown-body .pl-ba {
  color: #586069;
}

.emotion-2 .markdown-body .pl-sg {
  color: #959da5;
}

.emotion-2 .markdown-body .pl-corl {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #032f62;
}

.emotion-2 .markdown-body .octicon {
  display: inline-block;
  vertical-align: text-top;
  fill: currentColor;
}

.emotion-2 .markdown-body a {
  background-color: transparent;
}

.emotion-2 .markdown-body a:active,
.emotion-2 .markdown-body a:hover {
  outline-width: 0;
}

.emotion-2 .markdown-body strong {
  font-weight: inherit;
}

.emotion-2 .markdown-body strong {
  font-weight: bolder;
}

.emotion-2 .markdown-body img {
  border-style: none;
}

.emotion-2 .markdown-body code,
.emotion-2 .markdown-body kbd,
.emotion-2 .markdown-body pre {
  font-family: monospace,monospace;
  font-size: 1em;
}

.emotion-2 .markdown-body hr {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}

.emotion-2 .markdown-body input {
  font: inherit;
  margin: 0;
}

.emotion-2 .markdown-body input {
  overflow: visible;
}

.emotion-2 .markdown-body [type="checkbox"] {
  box-sizing: border-box;
  padding: 0;
}

.emotion-2 .markdown-body * {
  box-sizing: border-box;
}

.emotion-2 .markdown-body input {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.emotion-2 .markdown-body a {
  color: #0366d6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 .markdown-body a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 .markdown-body strong {
  font-weight: 600;
}

.emotion-2 .markdown-body hr {
  height: 0;
  margin: 15px 0;
  overflow: hidden;
  background: transparent;
  border: 0;
  border-bottom: 1px solid #dfe2e5;
}

.emotion-2 .markdown-body hr::before {
  display: table;
  content: "";
}

.emotion-2 .markdown-body hr::after {
  display: table;
  clear: both;
  content: "";
}

.emotion-2 .markdown-body table {
  border-spacing: 0;
  border-collapse: collapse;
}

.emotion-2 .markdown-body td,
.emotion-2 .markdown-body th {
  padding: 0;
}

.emotion-2 .markdown-body h1,
.emotion-2 .markdown-body h2,
.emotion-2 .markdown-body h3,
.emotion-2 .markdown-body h4,
.emotion-2 .markdown-body h5,
.emotion-2 .markdown-body h6 {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-2 .markdown-body p {
  margin-top: 0;
  margin-bottom: 10px;
}

.emotion-2 .markdown-body blockquote {
  margin: 0;
}

.emotion-2 .markdown-body ul,
.emotion-2 .markdown-body ol {
  padding-left: 0;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-2 .markdown-body ol ol,
.emotion-2 .markdown-body ul ol {
  list-style-type: lower-roman;
}

.emotion-2 .markdown-body ul ul ol,
.emotion-2 .markdown-body ul ol ol,
.emotion-2 .markdown-body ol ul ol,
.emotion-2 .markdown-body ol ol ol {
  list-style-type: lower-alpha;
}

.emotion-2 .markdown-body dd {
  margin-left: 0;
}

.emotion-2 .markdown-body code {
  font-family: "SFMono-Regular",Consolas,"Liberation Mono",Menlo,Courier,monospace;
  font-size: 12px;
}

.emotion-2 .markdown-body pre {
  margin-top: 0;
  margin-bottom: 0;
  font-family: "SFMono-Regular",Consolas,"Liberation Mono",Menlo,Courier,monospace;
  font-size: 12px;
}

.emotion-2 .markdown-body .octicon {
  vertical-align: text-bottom;
}

.emotion-2 .markdown-body .pl-0 {
  padding-left: 0 !important;
}

.emotion-2 .markdown-body .pl-1 {
  padding-left: 4px !important;
}

.emotion-2 .markdown-body .pl-2 {
  padding-left: 8px !important;
}

.emotion-2 .markdown-body .pl-3 {
  padding-left: 16px !important;
}

.emotion-2 .markdown-body .pl-4 {
  padding-left: 24px !important;
}

.emotion-2 .markdown-body .pl-5 {
  padding-left: 32px !important;
}

.emotion-2 .markdown-body .pl-6 {
  padding-left: 40px !important;
}

.emotion-2 .markdown-body::before {
  display: table;
  content: "";
}

.emotion-2 .markdown-body::after {
  display: table;
  clear: both;
  content: "";
}

.emotion-2 .markdown-body>*:first-child {
  margin-top: 0 !important;
}

.emotion-2 .markdown-body>*:last-child {
  margin-bottom: 0 !important;
}

.emotion-2 .markdown-body a:not([href]) {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 .markdown-body .anchor {
  float: left;
  padding-right: 4px;
  margin-left: -20px;
  line-height: 1;
}

.emotion-2 .markdown-body .anchor:focus {
  outline: none;
}

.emotion-2 .markdown-body p,
.emotion-2 .markdown-body blockquote,
.emotion-2 .markdown-body ul,
.emotion-2 .markdown-body ol,
.emotion-2 .markdown-body dl,
.emotion-2 .markdown-body table,
.emotion-2 .markdown-body pre {
  margin-top: 0;
  margin-bottom: 16px;
}

.emotion-2 .markdown-body hr {
  height: 0.25em;
  padding: 0;
  margin: 24px 0;
  background-color: #e1e4e8;
  border: 0;
}

.emotion-2 .markdown-body blockquote {
  padding: 0 1em;
  color: #6a737d;
  border-left: 0.25em solid #dfe2e5;
}

.emotion-2 .markdown-body blockquote>:first-child {
  margin-top: 0;
}

.emotion-2 .markdown-body blockquote>:last-child {
  margin-bottom: 0;
}

.emotion-2 .markdown-body kbd {
  display: inline-block;
  padding: 3px 5px;
  font-size: 11px;
  line-height: 10px;
  color: #444d56;
  vertical-align: middle;
  background-color: #fafbfc;
  border: solid 1px #c6cbd1;
  border-bottom-color: #959da5;
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 #959da5;
}

.emotion-2 .markdown-body h1,
.emotion-2 .markdown-body h2,
.emotion-2 .markdown-body h3,
.emotion-2 .markdown-body h4,
.emotion-2 .markdown-body h5,
.emotion-2 .markdown-body h6 {
  margin-top: 24px;
  margin-bottom: 16px;
  line-height: 1.25;
}

.emotion-2 .markdown-body h1 .octicon-link,
.emotion-2 .markdown-body h2 .octicon-link,
.emotion-2 .markdown-body h3 .octicon-link,
.emotion-2 .markdown-body h4 .octicon-link,
.emotion-2 .markdown-body h5 .octicon-link,
.emotion-2 .markdown-body h6 .octicon-link {
  color: #1b1f23;
  vertical-align: middle;
  visibility: hidden;
}

.emotion-2 .markdown-body h1:hover .anchor,
.emotion-2 .markdown-body h2:hover .anchor,
.emotion-2 .markdown-body h3:hover .anchor,
.emotion-2 .markdown-body h4:hover .anchor,
.emotion-2 .markdown-body h5:hover .anchor,
.emotion-2 .markdown-body h6:hover .anchor {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 .markdown-body h1:hover .anchor .octicon-link,
.emotion-2 .markdown-body h2:hover .anchor .octicon-link,
.emotion-2 .markdown-body h3:hover .anchor .octicon-link,
.emotion-2 .markdown-body h4:hover .anchor .octicon-link,
.emotion-2 .markdown-body h5:hover .anchor .octicon-link,
.emotion-2 .markdown-body h6:hover .anchor .octicon-link {
  visibility: visible;
}

.emotion-2 .markdown-body h1 {
  padding-bottom: 0.3em;
  font-size: 1.3em;
  border-bottom: 1px solid #eaecef;
}

.emotion-2 .markdown-body h2 {
  padding-bottom: 0.3em;
  font-size: 1.15em;
  border-bottom: 1px solid #eaecef;
}

.emotion-2 .markdown-body h3 {
  font-size: 1em;
}

.emotion-2 .markdown-body h4 {
  font-size: 0.875em;
}

.emotion-2 .markdown-body h5 {
  font-size: 0.85em;
}

.emotion-2 .markdown-body h6 {
  font-size: 0.8em;
  color: #6a737d;
}

.emotion-2 .markdown-body ul,
.emotion-2 .markdown-body ol {
  padding-left: 2em;
}

.emotion-2 .markdown-body ul ul,
.emotion-2 .markdown-body ul ol,
.emotion-2 .markdown-body ol ol,
.emotion-2 .markdown-body ol ul {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-2 .markdown-body li {
  word-wrap: break-all;
}

.emotion-2 .markdown-body li>p {
  margin-top: 16px;
}

.emotion-2 .markdown-body li+li {
  margin-top: 0.25em;
}

.emotion-2 .markdown-body dl {
  padding: 0;
}

.emotion-2 .markdown-body dl dt {
  padding: 0;
  margin-top: 16px;
  font-size: 1em;
  font-style: italic;
  font-weight: 600;
}

.emotion-2 .markdown-body dl dd {
  padding: 0 16px;
  margin-bottom: 16px;
}

.emotion-2 .markdown-body table {
  display: block;
  width: 100%;
  overflow: auto;
}

.emotion-2 .markdown-body table th {
  font-weight: 600;
}

.emotion-2 .markdown-body table th,
.emotion-2 .markdown-body table td {
  padding: 6px 13px;
  border: 1px solid rgba(0,0,0,0.3);
}

.emotion-2 .markdown-body table tr {
  background-color: #FAFAFA border-top:1px solid rgba(0,0,0,0.3);
}

.emotion-2 .markdown-body table tr:nth-child(2n) {
  background-color: rgba(0,0,0,0.3);
}

.emotion-2 .markdown-body img {
  max-width: 100%;
  box-sizing: content-box;
  background-color: #FAFAFA;
}

.emotion-2 .markdown-body img[align=right] {
  padding-left: 20px;
}

.emotion-2 .markdown-body img[align=left] {
  padding-right: 20px;
}

.emotion-2 .markdown-body code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.emotion-2 .markdown-body pre {
  word-wrap: normal;
}

.emotion-2 .markdown-body pre>code {
  padding: 0;
  margin: 0;
  font-size: 100%;
  word-break: normal;
  white-space: pre;
  background: transparent;
  border: 0;
}

.emotion-2 .markdown-body .highlight {
  margin-bottom: 16px;
}

.emotion-2 .markdown-body .highlight pre {
  margin-bottom: 0;
  word-break: normal;
}

.emotion-2 .markdown-body .highlight pre,
.emotion-2 .markdown-body pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #F5F5F5 border-radius:3px;
}

.emotion-2 .markdown-body pre code {
  display: inline;
  max-width: auto;
  padding: 0;
  margin: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
  background-color: transparent;
  border: 0;
}

.emotion-2 .markdown-body .full-commit .btn-outline:not(:disabled):hover {
  color: #005cc5;
  border-color: #005cc5;
}

.emotion-2 .markdown-body kbd {
  display: inline-block;
  padding: 3px 5px;
  font: 11px "SFMono-Regular",Consolas,"Liberation Mono",Menlo,Courier,monospace;
  line-height: 10px;
  color: rgba(0,0,0,.54);
  vertical-align: middle;
  background-color: #fafbfc;
  border: solid 1px #d1d5da;
  border-bottom-color: #c6cbd1;
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 #c6cbd1;
}

.emotion-2 .markdown-body:checked+.radio-label {
  position: relative;
  z-index: 1;
  border-color: #0366d6;
}

.emotion-2 .markdown-body .task-list-item {
  list-style-type: none;
}

.emotion-2 .markdown-body .task-list-item+.task-list-item {
  margin-top: 3px;
}

.emotion-2 .markdown-body .task-list-item input {
  margin: 0 0.2em 0.25em -1.6em;
  vertical-align: middle;
}

.emotion-2 .markdown-body hr {
  border-bottom-color: #eee;
}

.emotion-0 {
  font-size: 1rem;
  line-height: 1rem;
  position: relative;
  font-family: inherit;
  all: inherit;
}

<div
  class="emotion-2 emotion-3"
>
  <div
    class="markdown-body emotion-0 emotion-1"
  >
    <h1>
      Hello World
    </h1>
    

    <p>
      This is a test
    </p>
    

    <p>
      This is a 
      <strong>
        bold
      </strong>
       test
    </p>
    

    <p>
      This is a 
      <em>
        italic
      </em>
       test
    </p>
    

    <p>
      Should support inline latex 
      <span
        class="katex"
      >
        <span
          class="katex-mathml"
        >
          <math>
            <semantics>
              <mrow>
                <mn>
                  1
                </mn>
                <mo>
                  +
                </mo>
                <mn>
                  2
                </mn>
                <mo>
                  =
                </mo>
                <mn>
                  1
                </mn>
                <mn>
                  0
                </mn>
              </mrow>
              <annotation
                encoding="application/x-tex"
              >
                1 + 2 = 10
              </annotation>
            </semantics>
          </math>
        </span>
        <span
          aria-hidden="true"
          class="katex-html"
        >
          <span
            class="strut"
            style="height:0.64444em;"
          />
          <span
            class="strut bottom"
            style="height:0.72777em;vertical-align:-0.08333em;"
          />
          <span
            class="base"
          >
            <span
              class="mord"
            >
              1
            </span>
            <span
              class="mord rule"
              style="margin-right:0.2222222222222222em;"
            />
            <span
              class="mbin"
            >
              +
            </span>
            <span
              class="mord rule"
              style="margin-right:0.2222222222222222em;"
            />
            <span
              class="mord"
            >
              2
            </span>
            <span
              class="mord rule"
              style="margin-right:0.2777777777777778em;"
            />
            <span
              class="mrel"
            >
              =
            </span>
            <span
              class="mord rule"
              style="margin-right:0.2777777777777778em;"
            />
            <span
              class="mord"
            >
              1
            </span>
            <span
              class="mord"
            >
              0
            </span>
          </span>
        </span>
      </span>
    </p>
    

    <p>
      <span
        class="katex-display"
      >
        <span
          class="katex"
        >
          <span
            class="katex-mathml"
          >
            <math>
              <semantics>
                <mrow>
                  <mi>
                    x
                  </mi>
                  <mo>
                    =
                  </mo>
                  <msub>
                    <mi>
                      x
                    </mi>
                    <mn>
                      2
                    </mn>
                  </msub>
                </mrow>
                <annotation
                  encoding="application/x-tex"
                >
                  x = x_2

                </annotation>
              </semantics>
            </math>
          </span>
          <span
            aria-hidden="true"
            class="katex-html"
          >
            <span
              class="strut"
              style="height:0.43056em;"
            />
            <span
              class="strut bottom"
              style="height:0.58056em;vertical-align:-0.15em;"
            />
            <span
              class="base"
            >
              <span
                class="mord mathit"
              >
                x
              </span>
              <span
                class="mord rule"
                style="margin-right:0.2777777777777778em;"
              />
              <span
                class="mrel"
              >
                =
              </span>
              <span
                class="mord rule"
                style="margin-right:0.2777777777777778em;"
              />
              <span
                class="mord"
              >
                <span
                  class="mord mathit"
                >
                  x
                </span>
                <span
                  class="msupsub"
                >
                  <span
                    class="vlist-t vlist-t2"
                  >
                    <span
                      class="vlist-r"
                    >
                      <span
                        class="vlist"
                        style="height:0.30110799999999993em;"
                      >
                        <span
                          style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"
                        >
                          <span
                            class="pstrut"
                            style="height:2.7em;"
                          />
                          <span
                            class="sizing reset-size6 size3 mtight"
                          >
                            <span
                              class="mord mtight"
                            >
                              2
                            </span>
                          </span>
                        </span>
                      </span>
                      <span
                        class="vlist-s"
                      >
                        ​
                      </span>
                    </span>
                    <span
                      class="vlist-r"
                    >
                      <span
                        class="vlist"
                        style="height:0.15em;"
                      />
                    </span>
                  </span>
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </p>
    

  </div>
</div>
`;
